### PR TITLE
Simplifying functional event handling

### PR DESCRIPTION
**Pull request by jakearchibald**
I've simplified event firing as part of https://github.com/w3c/ServiceWorker/pull/1199.

Don't merge this until https://github.com/w3c/ServiceWorker/pull/1199 merges, but you can review now.

***
<a href="https://pr-preview.s3.amazonaws.com/jakearchibald/payment-handler/pull/305.html" title="Last updated on Jun 21, 2018, 3:44 PM GMT (aa5029c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-handler/305/16e9e5b...jakearchibald:aa5029c.html" title="Last updated on Jun 21, 2018, 3:44 PM GMT (aa5029c)">Diff</a>

### CODE REVIEW

**Comment by jakearchibald** [comment]
This can merge now btw. Anyone more familiar with the spec able to handle the conflicts?

File: index.html [hunk 1]
```diff
@@ -967,33 +967,47 @@ <h2>
...
+              <dt><a data-lt=
+              "PaymentRequestEvent.paymentRequestOrigin">paymentRequestOrigin</a>
+              </dt>
+              <dd><a data-cite=
+              "!HTML#ascii-serialisation-of-an-origin">the serialization of the
+              origin</a> of the context where PaymentRequest was initialized.
+              </dd>
+              <dt><a data-lt=
+              "PaymentRequestEvent.paymentRequestId">paymentRequestId</a></dt>
+              <dd>The [[\details]].<a>id</a> from the <a>PaymentRequest</a></dd>
```
Comment (rsolomakhin) [inline index.html:990]: Bad merge: please remove paymentRequestId from CanMakePaymentEvent.

File: index.html [hunk 2]
```diff
@@ -967,33 +967,47 @@ <h2>
...
+              <dt><a data-lt="PaymentRequestEvent.methodData">methodData</a>
+              </dt>
+              <dd>The result of executing the <a>MethodData Population
+              Algorithm</a></dd>
+              <dt><a data-lt="PaymentRequestEvent.modifiers">modifiers</a>
```
Comment (rsolomakhin) [inline index.html:995]: Bad merge: please remove modifiers from CanMakePaymentEvent.

File: index.html [hunk 3]
```diff
@@ -967,33 +967,47 @@ <h2>
...
+              </dt>
+              <dd>The result of executing the <a>Modifiers Population
+              Algorithm</a></dd>
+              <dt><a data-lt="PaymentRequestEvent.total">total</a>
```
Comment (rsolomakhin) [inline index.html:999]: Bad merge: please remove total from CanMakePaymentEvent.

File: index.html [hunk 4]
```diff
@@ -967,33 +967,47 @@ <h2>
...
+              </dt>
+              <dd>A <a>structured clone</a> of the total field on the
+              <a>PaymentDetailsInit</a> from the corresponding
+              <a>PaymentRequest</a>.</dd>
+              <dt><a data-lt=
```
Comment (rsolomakhin) [inline index.html:1004]: Bad merge: please remove instrumentKey from CanMakePaymentEvent.

**Review by rsolomakhin — CHANGES_REQUESTED** [review]
Thank you for the pull request! Please remove the extra fields that I pointed out from CanMakePaymentEvent.

**Comment by rsolomakhin** [comment]
@jakearchibald : Could you resolve conflicts so we can merge this, please? Thank you.

**Comment by rsolomakhin** [comment]
(If you don't have time, we can do it.)

**Comment by jakearchibald** [comment]
On travels right now. Gimmie a week 😄.

**Comment by marcoscaceres** [comment]
> (If you don't have time, we can do it.)

We should do it. Let poor Jake have a holiday :)

**Comment by ianbjacobs** [comment]
May be superseded by
https://github.com/w3c/payment-handler/pull/327

**Comment by ianbjacobs** [comment]
Hi @jakearchibald, closing this pull request. Thank you for doing it. With Rouslan we made your changes with a newer version of the spec:
 https://github.com/w3c/payment-handler/pull/327#pullrequestreview-165783247

**Pull request closed without merging**